staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -564,15 +564,26 @@ func (f *featureGate) featureSpec(key Feature) (FeatureSpec, error) {
 }
 
 func (f *featureGate) unsafeRecordQueried(key Feature) {
-	queriedFeatures := map[Feature]struct{}{}
-	for k := range f.queriedFeatures.Load().(map[Feature]struct{}) {
-		queriedFeatures[k] = struct{}{}
-	}
-	if _, ok := queriedFeatures[key]; ok {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+	queriedFeatures := f.loadQueriedFeatures()
+	if queriedFeatures.Has(key) {
 		return
 	}
-	queriedFeatures[key] = struct{}{}
-	f.queriedFeatures.Store(queriedFeatures)
+	// Clone queriedFeatures only when it needs to be mutated
+	queriedFeatures = sets.New[Feature]()
+	for k := range f.loadQueriedFeatures() {
+		queriedFeatures.Insert(k)
+	}
+	queriedFeatures.Insert(key)
```

Comment:
```suggestion
	newQueriedFeatures := queriedFeatures.Clone()
	newQueriedFeatures.Insert(Key)
```

Reply:
Done. Removed locking as a proper cloning is done now